src/slam_toolbox_common.cpp

```diff
@@ -319,8 +319,15 @@ bool SlamToolbox::shouldStartWithPoseGraph(
         pose.y = read_pose[1];
         pose.theta = read_pose[2];
       }
-    } else {
+    } else if (this->get_parameter("map_start_at_dock", start_at_dock)) {
       start_at_dock = this->get_parameter("map_start_at_dock").as_bool();
+    } else {
+      start_at_dock = false;
```

Comment:
Remove this, its not robust, if you didn't provide either of these parameters, it should fail and the function return false.

Reply:
This would also imply you want to fail if the number of parameters is defined wrong instead of using origin?

Reply:
Wouldn't it be better to throw an exception when configured badly instead of just returning false?

Reply:
No, if neither are set then its just a normal bring up into mapping mode